### ISSUE TITLE
Incorrect reportArgumentType on partial positional-only arguments

### ISSUE THREAD

**Issue by twoodwark** [microsoft/pyright#10954]
Type checking  on `partial` gives incorrect errors for some function signatures.

```python
from collections.abc import Callable
from functools import partial
from typing import Concatenate, reveal_type

# --- This is fine
def unwrapped(s: int, a: int, b: str) -> int: ...

reveal_type(unwrapped)  # Type of "unwrapped" is "(s: int, a: int, b: str) -> int"
partial(unwrapped, 1, 0, "")  # (no error)

# --- This is also fine
def wrapper[T, **P]() -> Callable[[Callable[P, T]], Callable[P, T]]: ...

@wrapper()
def wrapped(s: int, a: int, b: str) -> int: ...

reveal_type(wrapped)  # Type of "wrapped" is "(s: int, a: int, b: str) -> int"
partial(wrapped, 1, 0, "")  # (no error)

# --- This is incorrect, where / appears in the signature
def unwrapped_slash(s: int, /, a: int, b: str) -> int: ...

reveal_type(unwrapped_slash)  # Type of "unwrapped_slash" is "(s: int, /, a: int, b: str) -> int"
partial(unwrapped_slash, 1, 0, "")  # Argument of type "Literal['']" cannot be assigned to parameter "b" of type "int" in function "unwrapped_slash"
partial(unwrapped_slash, 1, "")  # (no error)


# --- This is also incorrect, where Concatenate creates a similar positional-only signature
def wrapper2[T, **P, S]() -> Callable[
    [Callable[Concatenate[S, P], T]], Callable[Concatenate[S, P], T]
]: ...  # ... wrapper2 does something with first argument

@wrapper2()
def wrapped_concat(s: int, a: int, b: str) -> int: ...

reveal_type(wrapped_concat)  # Type of "wrapped_concat" is "(int, a: int, b: str) -> int"

partial(wrapped_concat, 1, 0, "")  # Argument of type "Literal['']" cannot be assigned to parameter "b" of type "int" in function ""
```

observed in vs code pylance 2025.8.2, and pyright v1.1.405, Python 3.13.2

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in 1.1.406
